gnucash/gnome-utils/gnc-tree-container.hpp

```diff
@@ -74,6 +84,12 @@ template <typename ModelType>
 class GncTreeIter
 {
 public:
+    using iterator_category = std::forward_iterator_tag;
+    using value_type = ModelType;
+    using difference_type = std::ptrdiff_t;
+    using pointer = ModelType*;
+    using reference = ModelType&;
```

Comment:
This needs a comment like `/* Set iterator traits queried by STL algorithms. */` to protect it from deletion because the aliases aren't used anywhere in our code.